File: systock-api/src/main/java/io/github/joelytonneto/systock/model/entity/Venda.java
package io.github.joelytonneto.systock.model.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotEmpty;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Data
public class Venda {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@ManyToOne
	@JoinColumn(name = "id_cliente")
	private Cliente cliente;	
	
	@Column(name = "data_venda")
	@JsonFormat(pattern = "dd/MM/yyyy")
    private LocalDate dataVenda;
	
	@Column(nullable = false)
	@DecimalMin(value = "0", message = "{campo.valor.bruto.obrigatorio}")
	private BigDecimal valorBruto;
	
	@Column(nullable = false)
	@DecimalMin(value = "0", message = "{campo.desconto.obrigatorio}")
	private BigDecimal desconto;
	
	@Column(nullable = false)
	@DecimalMin(value = "0", message = "{campo.valor.liquido.obrigatorio}")
	private BigDecimal valorLiquido;
	
	@Column
	@DecimalMin(value = "0", message = "{campo.valor.pago.obrigatorio}")
	private BigDecimal valorPago;
	
	@Column
	@DecimalMin(value = "0", message = "{campo.valor.troco.obrigatorio}")
	private BigDecimal troco;
	
	@ManyToOne
	@JoinColumn(name = "id_entregador")
	private Entregador entregador;
	
	@Column
	@DecimalMin(value = "0", message = "{campo.taxa.entrega.obrigatorio}")
	private BigDecimal taxaEntrega;	
	
	@Column(columnDefinition="TEXT")
	private String observacao;	
	
	@Column(nullable = false, length = 50)    
    @NotEmpty(message = "{campo.status.pedido.obrigatorio}")    
	private String statusPedido;
	
	@Column(nullable = false)
	private boolean pago;

}

File: systock-app/src/app/vendas/venda.ts
import { Produto } from "../produtos/produto";
import { ItensVenda } from "./model/itensVenda";
import { Pagamento } from "./model/pagamento";
import { FormaPagamento } from './../formas-pagamentos/formaPagamento';
import { Cliente } from "../clientes/cliente";
import { Entregador } from "../entregadores/entregador";

export class Venda {
    id: number;
    idCliente: number;
    cliente: Cliente;
    produto: Produto;
    formaPagamento: FormaPagamento;
    dataVenda: string;
    valorBruto: number;
    desconto: number = 0;
    valorLiquido: number;
    valorPago: number;
    troco: number = 0;
    idEntregador: number;
    entregador: Entregador;
    taxaEntrega: number = 0;
    observacao: string;
    statusPedido: string;
    pago: boolean;
    itensVenda: Array<ItensVenda> = [];
    pagamentosVenda: Array<Pagamento> = [];
    enderecoVenda: Cliente = new Cliente();
}